command/meta_providers.go

```diff
@@ -399,20 +399,23 @@ func devOverrideProviderFactory(provider addrs.Provider, localDir getproviders.P
 // running, and implements providers.Interface against it.
 func unmanagedProviderFactory(provider addrs.Provider, reattach *plugin.ReattachConfig) providers.Factory {
 	return func() (providers.Interface, error) {
-
 		config := &plugin.ClientConfig{
 			HandshakeConfig:  tfplugin.Handshake,
 			Logger:           logging.NewProviderLogger("unmanaged."),
 			AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
 			Managed:          false,
 			Reattach:         reattach,
 		}
-		// TODO: we probably shouldn't hardcode the protocol version
-		// here, but it'll do for now, because only one protocol
-		// version is supported. Eventually, we'll probably want to
-		// sneak it into the JSON ReattachConfigs.
-		if plugins, ok := tfplugin.VersionedPlugins[5]; !ok {
-			return nil, fmt.Errorf("no supported plugins for protocol 5")
+
+		if reattach.ProtocolVersion == 0 {
+			// As of the 0.15 release, sdk.v2 doesn't include the protocol
+			// version in the ReattachConfig (only recently added to
+			// go-plugin), so client.NegotiatedVersion() always returns 0. We
+			// assume that an unmanaged provider reporting protocol version 0 is
+			// actually using proto v5 for backwards compatibility.
+			config.Plugins = tfplugin.VersionedPlugins[5]
+		} else if plugins, ok := tfplugin.VersionedPlugins[reattach.ProtocolVersion]; !ok {
```

Comment:
This doesn't properly check that a plugin exists in `VersionedPlugins` under key `5` and handle it gracefully if it doesn't.

```suggestion
			reattach.ProtocolVersion = 5
		}
		if plugins, ok := tfplugin.VersionedPlugins[reattach.ProtocolVersion]; !ok {
```

Reply:
Thank you, fixed!